### PR TITLE
219 refactor workflow automation to use berkley metadata models

### PR DESCRIPTION
**Pull request by mbthornton-lbl**
This PR provides updates to the `workflow_automation` framework to be compatible with Berkley metadata models and runtime endpoints.

This PR addresses several sub-issues:

Workflow Configuration
#237

### CODE REVIEW

File: nmdc_automation/config/workflows/workflows-mt.yaml [hunk 1]
```diff
@@ -57,30 +57,30 @@ Workflows:
...
 
   - Name: Metatranscriptome Reads QC Interleave
-    Type: nmdc:ReadQcAnalysisActivity
+    Type: nmdc:ReadQcAnalysis
     Enabled: True
     Analyte Category: Metatranscriptome
     Git_repo: https://github.com/microbiomedata/metaT_ReadsQC
-    Version: v0.0.7
-    Collection: read_qc_analysis_activity_set
+    Version: v0.0.3
```
Comment (aclum) [inline nmdc_automation/config/workflows/workflows-mt.yaml:64]: merge conflict kept the wrong version, this should be v0.0.7

File: nmdc_automation/config/workflows/workflows.yaml [hunk 2]
```diff
@@ -285,52 +285,47 @@ Workflows:
 
 
   - Name: MAGs
-    Type: nmdc:MagsAnalysisActivity
+    Type: nmdc:MagsAnalysis
     Enabled: True
     Analyte Category: Metagenome
     Git_repo: https://github.com/microbiomedata/metaMAGs
-    Version: v1.3.10
+    Version: v1.3.4
```
Comment (aclum) [inline nmdc_automation/config/workflows/workflows.yaml:292]: same here, merge conflict did not pick the right version, this should be v1.3.10

File: nmdc_automation/config/workflows/workflows.yaml
Comment (aclum) [inline nmdc_automation/config/workflows/workflows.yaml:1]: Merge resolution in at least this workflow config is not correct. Changes to Data Objects and versions should keep what is in main. Changes for class names, collections, and references to activity should come from the berkeley branches.

File: nmdc_automation/workflow_automation/models.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    name: str
+    collection: str
+    enabled: bool
+    analyte_category: str
+    filter_output_objects: List[str]
+    # TODO should type be optional?
+    type: Optional[str] = None
+
+    # workflow repository information
+    git_repo: Optional[str] = None
```
Comment (aclum) [inline nmdc_automation/workflow_automation/models.py:158]: git_repo is not an optional slot for this class.

File: nmdc_automation/config/workflows/workflows-mt.yaml (hunk 1 above)
Comment (mbthornton-lbl) [reply nmdc_automation/config/workflows/workflows-mt.yaml]: https://github.com/microbiomedata/nmdc_automation/pull/253/commits/01235e56038232ac22c09a20944a3b449210a258

File: nmdc_automation/workflow_automation/wfutils.py [hunk 4]
```diff
@@ -345,41 +344,30 @@ def activity_map(self):
...
-                nmdc.MetatranscriptomeAssembly,
+            "nmdc:MetagenomeAnnotation": (
+                self.nmdc_db.workflow_execution_set,
+                nmdc.MetagenomeAnnotation,
             ),
-            "nmdc:MetagenomeAnnotationActivity": (
-                self.nmdc_db.metagenome_annotation_activity_set,
-                nmdc.MetagenomeAnnotationActivity,
-            ),
-            "nmdc:MetatranscriptomeAnnotationActivity": (
```
Comment (aclum) [inline nmdc_automation/workflow_automation/wfutils.py:372]: missing [MetatranscriptomeAnnotation](https://microbiomedata.github.io/berkeley-schema-fy24/MetatranscriptomeAnnotation/), [MetatranscriptomeExpressionAnalysis](https://microbiomedata.github.io/berkeley-schema-fy24/MetatranscriptomeExpressionAnalysis/), [MetatranscriptomeAssembly](https://microbiomedata.github.io/berkeley-schema-fy24/MetatranscriptomeAssembly/)

File: nmdc_automation/config/workflows/workflows.yaml (hunk 2 above)
Comment (mbthornton-lbl) [reply nmdc_automation/config/workflows/workflows.yaml]: updated

File: tests/fixtures/mags_activity.json
Comment (aclum) [inline tests/fixtures/mags_activity.json:1]: data objects in this test json use legacy style identifiers.

Comment (aclum) [reply tests/fixtures/mags_activity.json:1]: Do we need this file? we have tests/fixtures/mags_record.json which has updated identifiers.

File: tests/import_test.yaml
Comment (aclum) [inline tests/import_test.yaml:1]: should this file still exist? I thought you updated to using the prod cofigs for testing.

File: tests/test_imports.py [hunk 5]
```diff
@@ -29,8 +29,8 @@ def test_activity_mapper(gold_mapper):
     gold_mapper.activity_mapper()
     gold_mapper.post_nmdc_database_object()
     db = gold_mapper.get_database_object_dump()
-    assert len(db.metagenome_annotation_activity_set) == 1
-    assert len(db.mags_activity_set) == 1
+    #TODO I think this assert statement should be 2 ANC
```
Comment (aclum) [inline tests/test_imports.py:32]: @mbthornton-lbl  where you able to double check the expected number here?

File: tests/workflows_test.yaml
Comment (aclum) [inline tests/workflows_test.yaml:1]: based on previous discussions I don't think this file should exist anymore.

File: tests/workflows_test2.yaml
Comment (aclum) [inline tests/workflows_test2.yaml:1]: same, should this file still exist?

**Review by aclum — CHANGES_REQUESTED** [review]
See individual comments.

File: nmdc_automation/workflow_automation/wfutils.py (hunk 4 above)
Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/wfutils.py:372]: https://github.com/microbiomedata/nmdc_automation/pull/253/commits/bb546e3eb31b0927465535e98d5b84ed6135e019

File: tests/fixtures/mags_activity.json
Comment (mbthornton-lbl) [reply tests/fixtures/mags_activity.json:1]: https://github.com/microbiomedata/nmdc_automation/pull/253/commits/dbd715fa8d71406d782dd228621a3bfac06d8cac

File: tests/import_test.yaml
Comment (mbthornton-lbl) [reply tests/import_test.yaml:1]: Only the workflow configs.  Right now changing to the prod `import.yaml` causes unit test failures 🙁
 `test_imports.py` has a lot of bare-number assertions such as:
```
    #TODO I think this assert statement should be 2 ANC
    assert len(db.workflow_execution_set) == 4
    assert len(db.data_object_set) == 3
``` 
I don't have a good understanding of the import code yet, so it's difficult for me to determine what correct expected values should be.

TL;DR. It should not exist, but removing it now is problematic

Comment (mbthornton-lbl) [reply tests/import_test.yaml:1]: I made this item do address the `import_automation` code and tests.
https://github.com/microbiomedata/nmdc_automation/issues/254

File: tests/test_imports.py (hunk 5 above)
Comment (mbthornton-lbl) [reply tests/test_imports.py:32]: No, and I don't really understand the import code well enough yet to know what sensible expect values should be.
In this case if we use the prod `imports.yaml` we get different results: 4 vs. 2

I would prefer to deal with imports outside of the scope of this PR
https://github.com/microbiomedata/nmdc_automation/issues/254

File: tests/workflows_test.yaml
Comment (mbthornton-lbl) [reply tests/workflows_test.yaml:1]: Gone

File: tests/workflows_test2.yaml
Comment (mbthornton-lbl) [reply tests/workflows_test2.yaml:1]: Gone

File: nmdc_automation/workflow_automation/models.py (hunk 3 above)
Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/models.py:163]: Sequencing workflows don't have a `git_repo`

Comment (aclum) [reply nmdc_automation/workflow_automation/models.py:163]: My strong preference is to remove Sequencing Interleaved and Sequencing Noninterleaved from the configs so this isn't an issue. This would also allow us to make version required on our end even though it isn't required by the schema currently. Workaround option is to add git_url: https://github.com/microbiomedata/RawSequencingData to Sequencing Interleaved and Sequencing Noninterleaved, that is what the existing mongo records use. This is a repo Shane made to deal with the requirement on the schema end.

File: nmdc_automation/workflow_automation/models.py [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+        return self.process.has_input
+
+    @property
+    def has_output(self):
+        return self.process.has_output
+
+    @property
+    def git_url(self):
+        """ workflow executions have a git_url field, data generations do not"""
+        default_url = "http://github.com/microbiomedata"
```
Comment (aclum) [inline nmdc_automation/workflow_automation/models.py:95]: This is not specific enough, don't use this to pull the actual wdl?

File: nmdc_automation/workflow_automation/models.py (hunk 3 above)
Comment (aclum) [reply nmdc_automation/workflow_automation/models.py:163]: Why was this comment resolved?

File: nmdc_automation/workflow_automation/models.py (hunk 6 above)
Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/models.py:95]: We do return the actual url when there is one:
`getattr(self.process, "git_url", default_url)` = Get the `git_url` attribute if it is set, otherwise return the default.

Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/models.py:95]: `self.process` can hold either a WorkflowExecution or a DataGeneration, which does not have a Git url associated with it

Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/models.py:95]: I changed this to return `None` if there is no `git_url` set, and all the tests pass.

File: nmdc_automation/config/workflows/workflows.yaml
Comment (aclum) [reply nmdc_automation/config/workflows/workflows.yaml:1]: This is still not resolved. The versions are correct but the merge conflict for the data objects still is not correct.  (ie data_object_type: Metagenome LQ Bins Compression File) 

Also Workflow Execution stats  that came from main should be kept 
Ie binned_contig_num: "{outputs.final_stats_json.binned_contig_num}" should be in this file

File: nmdc_automation/config/workflows/workflows.yaml [hunk 7]
```diff
@@ -328,9 +328,9 @@ Workflows:
       description: CheckM for {id}
       name: CheckM statistics report
     - output: final_hqmq_bins_zip
-      data_object_type: Metagenome HQMQ Bins Compression File
```
Comment (aclum) [inline nmdc_automation/config/workflows/workflows.yaml:331]: This is backwards, keep Metagenome HQMQ Bins Compression File, remove Metagenome Bins

**Review by aclum — APPROVED** [review]

File: nmdc_automation/config/workflows/workflows.yaml (hunk 7 above)
Comment (aclum) [reply nmdc_automation/config/workflows/workflows.yaml]: resolved by [696c8f6](https://github.com/microbiomedata/nmdc_automation/pull/253/commits/696c8f6ad69c2ee478295bdca2aec3e952adec46)
